tests/Collection/BuilderCollectionFunctionalTest.php

```diff
@@ -0,0 +1,270 @@
+        $this->assertEquals([2], $result);
+    }
+
+    public function testFind(): void
+    {
+        $results = $this->collection->find(Query::query(x: Query::gt(1)))->toArray();
+        $this->assertCount(2, $results);
+        $this->assertEquals(2, $results[0]->x);
+    }
+
+    public function testFindOne(): void
+    {
+        $result = $this->collection->findOne(Query::query(x: Query::eq(1)));
+        $this->assertEquals(1, $result->x);
+    }
+
+    public function testFindOneAndDelete(): void
+    {
+        $result = $this->collection->findOneAndDelete(Query::query(x: Query::eq(1)));
+        $this->assertEquals(1, $result->x);
+
+        $result = $this->collection->find()->toArray();
+        $this->assertCount(2, $result);
+    }
+
+    public function testFindOneAndReplace(): void
+    {
+        $this->collection->insertOne(['x' => 1]);
+
+        $result = $this->collection->findOneAndReplace(
+            Query::query(x: Query::lt(2)),
+            ['x' => 3],
+        );
+        $this->assertEquals(1, $result->x);
+
+        $result = $this->collection->findOne(Query::query(x: Query::eq(3)));
+        $this->assertEquals(3, $result->x);
+    }
+
+    public function testFindOneAndUpdate(): void
+    {
+        $result = $this->collection->findOneAndUpdate(
+            Query::query(x: Query::lt(2)),
+            // @todo Use Builder when update operators are supported by PHPLIB-1507
+            ['$set' => ['x' => 3]],
+        );
+        $this->assertEquals(1, $result->x);
+
+        $result = $this->collection->findOne(Query::query(x: Query::eq(3)));
+        $this->assertEquals(3, $result->x);
+    }
+
+    public function testReplaceOne(): void
+    {
+        $this->collection->insertOne(['x' => 1]);
+
+        $result = $this->collection->replaceOne(
+            Query::query(x: Query::lt(2)),
+            ['x' => 3],
+        );
+        $this->assertEquals(1, $result->getModifiedCount());
+
+        $result = $this->collection->findOne(Query::query(x: Query::eq(3)));
+        $this->assertEquals(3, $result->x);
+    }
+
+    public function testUpdateOne(): void
+    {
+        $this->collection->insertOne(['x' => 1]);
+
+        $result = $this->collection->updateOne(
+            Query::query(x: Query::lt(2)),
+            // @todo Use Builder when update operators are supported by PHPLIB-1507
+            ['$set' => ['x' => 3]],
+        );
+        $this->assertEquals(1, $result->getModifiedCount());
+
+        $result = $this->collection->findOne(Query::query(x: Query::eq(3)));
+        $this->assertEquals(3, $result->x);
+    }
+
+    public function testUpdateWithPipeline(): void
+    {
+        $result = $this->collection->updateOne(
+            Query::query(x: Query::lt(2)),
+            new Pipeline(
+                Stage::set(x: 3),
+            ),
+        );
+
+        $this->assertEquals(1, $result->getModifiedCount());
+    }
```

Comment:
This test is failing in evergreen:

```
[2024/09/10 23:21:34.399] ERROR:  (MongoDB\Driver\Exception\BulkWriteException)
[2024/09/10 23:21:34.399] MongoDB\Tests\Collection\BuilderCollectionFunctionalTest::testUpdateWithPipeline
[2024/09/10 23:21:34.399] MongoDB\Driver\Exception\BulkWriteException: BSON field 'update.updates.u' is the wrong type 'array', expected type 'object'
[2024/09/10 23:21:34.399] /data/mci/b14552e6624c454125590fb9d49d9cc2/src/src/Operation/Update.php:191
[2024/09/10 23:21:34.399] /data/mci/b14552e6624c454125590fb9d49d9cc2/src/src/Operation/UpdateOne.php:107
[2024/09/10 23:21:34.399] /data/mci/b14552e6624c454125590fb9d49d9cc2/src/src/Collection.php:1070
[2024/09/10 23:21:34.399] /data/mci/b14552e6624c454125590fb9d49d9cc2/src/tests/Collection/BuilderCollectionFunctionalTest.php:212
```

Reply:
What versions is this failing on? Support for update pipelines was only added in 4.2.0, so you'll have to [skip this on older versions](https://github.com/mongodb/mongo-php-library/blob/5ebc5dc0bc3c5643b42d68e001bb797eebd36acc/tests/Operation/BulkWriteFunctionalTest.php#L204).